### PR TITLE
JA: add site/ja/tutorials/customization/custom_training_walkthrough.ipynb

### PR DESCRIPTION
**Pull request by masa-ita**

### CODE REVIEW

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 1]
```diff
@@ -0,0 +1,1139 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "Zx7wc0LuuxaJ"
+   },
+   "source": [
+    "## アヤメ分類問題\n",
+    "\n",
+    "あなたが植物学者で、発見したアヤメの花を分類する自動的な方法を探しているとします。機械学習には花を統計的に分類するための方法がたくさんあります。たとえば、高度な機械学習プログラムを使うと、写真を元に花を分類することができます。私達の願望はもう少し控えめです。アヤメの花を [sepals（萼片）](https://en.wikipedia.org/wiki/Sepal) と [petals（花弁）](https://en.wikipedia.org/wiki/Petal) の長さと幅を使って分類することにしましょう。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [weak suggestion] `方法` -> `アルゴリズム` . In my opinion, `アルゴリズム` is frequently used in Japanese.
[weak suggestion] `高度な` -> `洗練された`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 2]
```diff
@@ -0,0 +1,1139 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "kQhzD6P-uBoq"
+   },
+   "source": [
+    "上記のようにデータセットを見ることで次のことがわかります。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [weak suggestion] `上記のようにデータセットを見ることで` -> `このデータセットを見ることで`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 3]
```diff
@@ -0,0 +1,1139 @@
...
+    "\n",
+    "1. 最初の行はデータセットに関する情報を含むヘッダである\n",
+    "  * 全体で120のサンプルがある。各サンプルには4つの特徴量があり、3つのラベルの1つが当てはまる。\n",
+    "2. 続く行はデータレコードで、1行が1つの [*サンプル(example)*](https://developers.google.com/machine-learning/glossary/#example) である。ここでは、\n",
+    "  * 最初の4つのフィールドはサンプルの特徴を示す [*特徴量(features)*](https://developers.google.com/machine-learning/glossary/#feature) である。ここで各フィールドは花の計測値を表す浮動小数点値をもつ。\n",
+    "  * 最後の列は予想の対象となる [*ラベル(label)*](https://developers.google.com/machine-learning/glossary/#label) である。このデータセットでは、花の名前に対応する整数値、0、1、2 のいずれかである。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [suggestion] `最後の列は予想の対象となる` -> `最後の列は予測したい`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 4]
```diff
@@ -0,0 +1,1139 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "CCtwLoJhhDNc"
+   },
+   "source": [
+    "ラベルはそれぞれ文字列の名前（たとえば、\"setosa\"）に関連付けられていますが、機械学習は大抵の場合、数値に依存しています。ラベルの数値は名前付きの表現にマップされます。たとえば次のようになります。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [weak suggestion] `名前付きの` -> `名前の`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 5]
```diff
@@ -0,0 +1,1139 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "dqPkQExM2Pwt"
+   },
+   "source": [
+    "### `tf.data.Dataset` の作成\n",
+    "\n",
+    "TensorFlow の [Dataset API](https://www.tensorflow.org/guide/datasets) はデータのモデルへのロードについて、多くの一般的なケースを扱います。この API は、データを読み込んで訓練に使われる形態へ変換するための高レベル API です。詳しくは、 [Datasets Quick Start guide](https://www.tensorflow.org/get_started/datasets_quickstart) を参照してください。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [weak suggestion] `形態` -> `形式`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 6]
```diff
@@ -0,0 +1,1139 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "E63mArnQaAGz"
+   },
+   "source": [
+    "「特徴量のようなもの」はグループ化、言い換えると *バッチ化* されることに注意してください。それぞれのサンプル行のフィールドは、対応する特徴量の配列に追加されます。これらの特徴量配列に保存されるサンプルの数を設定するには、 `batch_size` を変更します。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [suggestion] `特徴量の配列` -> `特徴量配列`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 7]
```diff
@@ -0,0 +1,1139 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "NLy0Q1xCldVO"
+   },
+   "source": [
+    "この `Dataset` の特徴量要素は、いまは形状 `(batch_size, num_features)` の配列になっています。 最初のサンプルをいくつか見てみましょう。"
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [suggestion]
 ```
いまは形状 `(batch_size, num_features)`
```
 ->
```
`(batch_size, num_features)` の形状をした
```

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 8]
```diff
@@ -0,0 +1,1139 @@
...
+    "\n",
+    "機械学習を*使わずに*これらの4つの特徴量とアヤメの品種の関係を決定することは可能でしょうか？言い換えると、従来のプログラミング技術（たとえば、たくさんの条件文）を使ってモデルを作ることは可能でしょうか？おそらく、十分に時間をかけてデータセットを分析して、萼片と花弁の計測値と特定の品種との関係を決定すれば可能でしょう。もっと複雑なデータセットの場合には、これは困難であり、おそらく不可能です。機械学習を使ったよいアプローチでは、*あなたに代わってモデルを決定*してくれます。適切なタイプの機械学習モデルに、十分な量の典型的なサンプルを投入すれば、プログラムがあなたのためにこの関係性をみつけ出してくれるでしょう。\n",
+    "\n",
+    "###  モデルの選択\n",
+    "\n",
+    "訓練すべきモデルの種類を決める必要があります。モデルにはたくさんの種類があり、よいものを選択するには経験が必要です。このチュートリアルでは、アヤメの分類問題を解くために、ニューラルネットワークを使用します。[*ニューラルネットワーク(Neural networks)*](https://developers.google.com/machine-learning/glossary/#neural_network) は、特徴量とラベルの間の複雑な関係をみつけることができます。ニューラルネットワークは、1つかそれ以上の [*隠れ層(hidden layers)*](https://developers.google.com/machine-learning/glossary/#hidden_layer) に構成された高度なグラフ構造です。それぞれの隠れ層には1つかそれ以上の [*ニューロン(neurons)*](https://developers.google.com/machine-learning/glossary/#neuron) があります。ニューラルネットワークにはいくつかのカテゴリーがありますが、このプログラムでは、密な、あるいは、 [*全結合ニューラルネットワーク(fully-connected neural network)*](https://developers.google.com/machine-learning/glossary/#fully_connected_layer) を使用します。全結合ニューラルネットワークでは、1つの層の中のニューロンすべてが、前の層の*すべての*ニューロンからの入力を受け取ります。例として、図2に入力層、2つの隠れ層、そして、出力層からなる密なニューラルネットワークを図示します。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [suggestion] `１つかそれ以上の` -> `１つ以上の`
[weak suggestion] `に構成された高度なグラフ構造です` -> `で構成された高度なグラフ構造です`
[weak suggestion] `図示します` -> `示します`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 9]
```diff
@@ -0,0 +1,1139 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "W23DIMVPQEBt"
+   },
+   "source": [
+    "### Keras を使ったモデル構築\n",
+    "\n",
+    "TensorFlow の [tf.keras](https://www.tensorflow.org/api_docs/python/tf/keras) API は、モデルと層を作成するための好ましい方法です。Keras がすべてを結びつけるという複雑さを引き受けてくれるため、モデルや実験の構築がかんたんになります。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [weak suggestion] `好ましい` -> `おすすめの`

**Comment by lamberta** [comment]
The file `site/ja/tutorials/customization/eager/custom_training_walkthrough.ipynb` should be moved to deleted

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 1 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks．Fixed.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 2 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks．I'd like to change it as ”こうしてデータセットを見ることで".

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 3 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks.  It's better.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 4 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks．Changed.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 5 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks．Done.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 6 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks．Fixed.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 7 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thank you．It's better.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 8 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks.  Fixed.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 9 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks.  It's natural.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 10]
```diff
@@ -0,0 +1,1139 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "lOxFimtlKruu"
+   },
+   "source": [
+    "### オプティマイザーの作成\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [suggestion] `オプティマイザー` -> `オプティマイザ`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 11]
```diff
@@ -0,0 +1,1139 @@
...
+    "\n",
+    "[*オプティマイザー(optimizer)*](https://developers.google.com/machine-learning/crash-course/glossary#optimizer) は、`loss` 関数を最小化するため、計算された勾配をモデルの変数に適用します。損失関数は、曲面として考えることができ（図3 参照）、歩き回ることによって最小となる点をみつけたいのです。勾配は、一番急な上りの方向を示すため、逆方向に進んで丘を下ることになります。バッチごとに損失と勾配を計算することにより、訓練中のモデルを調節します。モデルは徐々に、損失を最小化する重みとバイアスの最適な組み合わせをみつけます。損失が小さいほど、モデルの予測がよくなります。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [suggestion] `バッチごとに損失と勾配を計算することにより` -> `バッチごとに損失と勾配を繰り返し計算することにより`
`iteratively` is missing in Japanese

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 12]
```diff
@@ -0,0 +1,1139 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "JtEZ1pCPn--z"
+   },
+   "source": [
+    "# カスタム訓練：ウォークスルー"
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [weak suggestion] `訓練` -> `学習` based on [this discussion](https://github.com/tfug/proofreading/issues/31#issuecomment-535847224)

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 13]
```diff
@@ -0,0 +1,1139 @@
...
+    "  <tr><td>\n",
+    "    <img src=\"https://cs231n.github.io/assets/nn3/opt1.gif\" width=\"70%\"\n",
+    "         alt=\"Optimization algorithms visualized over time in 3D space.\">\n",
+    "  </td></tr>\n",
+    "  <tr><td align=\"center\">\n",
+    "    <b>図3.</b> 3次元空間における最適化アルゴリズムの時系列可視化。<br/>(Source: <a href=\"http://cs231n.github.io/neural-networks-3/\">Stanford class CS231n</a>, MIT License, Image credit: <a href=\"https://twitter.com/alecrad\">Alec Radford</a>)\n",
+    "  </td></tr>\n",
+    "</table>\n",
+    "\n",
+    "TensorFlow には、訓練に使える [最適化アルゴリズム(optimization algorithms)](https://www.tensorflow.org/api_guides/python/train) がたくさんあります。このモデルでは、[*確率的勾配降下法(stochastic gradient descent)*](https://developers.google.com/machine-learning/crash-course/glossary#gradient_descent) (SGD) を実装した [tf.train.GradientDescentOptimizer](https://www.tensorflow.org/api_docs/python/tf/train/GradientDescentOptimizer) アルゴリズムを使用します。`learning_rate` （学習率）は、イテレーションごとに丘を下る際のステップのサイズを設定します。通常これはよりよい結果を得るために調整する *ハイパーパラメータ(hyperparameter)* です。 "
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [weak suggestion] `このモデルでは、[*確率的勾配降下法(stochastic gradient descent)*](https://developers.google.com/machine-learning/crash-course/glossary#gradient_descent) (SGD) を実装した [tf.train.GradientDescentOptimizer](https://www.tensorflow.org/api_docs/python/tf/train/GradientDescentOptimizer) アルゴリズムを使用します` -> `このモデルでは、[*確率的勾配降下法(stochastic gradient descent)*](https://developers.google.com/machine-learning/crash-course/glossary#gradient_descent) (SGD) アルゴリズムを実装した [tf.train.GradientDescentOptimizer](https://www.tensorflow.org/api_docs/python/tf/train/GradientDescentOptimizer) を使用します`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 14]
```diff
@@ -0,0 +1,1139 @@
...
+    "colab_type": "text",
+    "id": "7Y2VSELvwAvW"
+   },
+   "source": [
+    "### 訓練ループ\n",
+    "\n",
+    "すべての部品が揃ったので、モデルの訓練ができるようになりました。訓練ループは、モデルにデータセットのサンプルを供給し、モデルがよりよい予測を行えるようにします。下記のコードブロックは、この訓練のステップを構成します。\n",
+    "\n",
+    "1. *epoch（エポック）* をひとつずつ繰り返します。エポックとは、データセットをひととおり処理するということです。\n",
+    "2. エポック内では、訓練用の `Dataset（データセット）` のサンプルひとつずつから、その *features（特徴量）* (`x`) と *label（ラベル）* (`y`) を取り出して処理します。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [suggestion] `処理します` -> `繰り返し処理します`

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 15]
```diff
@@ -0,0 +1,1139 @@
...
+   "metadata": {
+    "colab_type": "text",
+    "id": "Zg8GoMZhLpGH"
+   },
+   "source": [
+    "## モデルの有効性評価\n",
+    "\n",
+    "モデルの訓練が終わったら、その性能の指標を得ることができます。\n",
+    "\n",
+    "*評価*とは、モデルの予測が効果的であるかどうかを決定することを言います。アヤメの分類でモデルの有効性を見定めるには、モデルに萼片と花弁の測定値をいくつか与え、どのアヤメの品種であるかを予測させます。そして、モデルの予測と実際のラベルを比較します。たとえば、あるモデルが入力サンプルの半分で正解の品種をえらんだとすると、[*正解率(accuracy)*](https://developers.google.com/machine-learning/glossary/#accuracy) は `0.5` ということになります。図４で示すモデルはもう少し効果的であり、5つの予測中4つで正解し、正解率は 80% です。\n",
```
Comment (kiszk) [inline site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: [weak suggestion] `効果的で` -> `どれだけ効果的で`

File: site/ja/tutorials/customization/eager/custom_training_walkthrough.ipynb [hunk 16]
```diff
@@ -0,0 +1,1139 @@
+{
```
Comment (AseiSugiyama) [inline site/ja/tutorials/customization/eager/custom_training_walkthrough.ipynb]: Could you delete this file from PR?

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 10 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks.  Fixed.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 11 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thank you．Fixed.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 12 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: I'd like to leave this as is now.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 13 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thank you.  It's natural.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 14 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks you.  I changed.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb (hunk 15 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks.  It's better.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 17]
```diff
@@ -0,0 +1,1139 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "bfV2Dai0Ow2o"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install tensorflow==2.0.0-beta1\n",
```
[the remark this answers is not in the document]
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Oops! Thank you.

File: site/ja/tutorials/customization/custom_training_walkthrough.ipynb [hunk 18]
```diff
@@ -0,0 +1,1139 @@
...
+    "\n",
+    "機械学習を*使わずに*これらの4つの特徴量とアヤメの品種の関係を決定することは可能でしょうか？言い換えると、従来のプログラミング技術（たとえば、たくさんの条件文）を使ってモデルを作ることは可能でしょうか？おそらく、十分に時間をかけてデータセットを分析して、萼片と花弁の計測値と特定の品種との関係を決定すれば可能でしょう。もっと複雑なデータセットの場合には、これは困難であり、おそらく不可能です。機械学習を使ったよいアプローチでは、*あなたに代わってモデルを決定*してくれます。適切なタイプの機械学習モデルに、十分な量の典型的なサンプルを投入すれば、プログラムがあなたのためにこの関係性をみつけ出してくれるでしょう。\n",
+    "\n",
+    "###  モデルの選択\n",
+    "\n",
+    "訓練すべきモデルの種類を決める必要があります。モデルにはたくさんの種類があり、よいものを選択するには経験が必要です。このチュートリアルでは、アヤメの分類問題を解くために、ニューラルネットワークを使用します。[*ニューラルネットワーク(Neural networks)*](https://developers.google.com/machine-learning/glossary/#neural_network) は、特徴量とラベルの間の複雑な関係をみつけることができます。ニューラルネットワークは、1つかそれ以上の [*隠れ層(hidden layers)*](https://developers.google.com/machine-learning/glossary/#hidden_layer) で構成された高度なグラフ構造です。それぞれの隠れ層には1つ以上の [*ニューロン(neurons)*](https://developers.google.com/machine-learning/glossary/#neuron) があります。ニューラルネットワークにはいくつかのカテゴリーがありますが、このプログラムでは、密な、あるいは、 [*全結合ニューラルネットワーク(fully-connected neural network)*](https://developers.google.com/machine-learning/glossary/#fully_connected_layer) を使用します。全結合ニューラルネットワークでは、1つの層の中のニューロンすべてが、前の層の*すべての*ニューロンからの入力を受け取ります。例として、図2に入力層、2つの隠れ層、そして、出力層からなる密なニューラルネットワークを示します。\n",
```
[the remark this answers is not in the document]
Comment (masa-ita) [reply site/ja/tutorials/customization/custom_training_walkthrough.ipynb]: Thanks.  I'd like to change it to "このプログラムでは、 [*全結合ニューラルネットワーク(fully-connected neural network または dense neural network)*](https://developers.google.com/machine-learning/glossary/#fully_connected_layer) を使用します。"

File: site/ja/tutorials/customization/eager/custom_training_walkthrough.ipynb (hunk 16 above)
Comment (masa-ita) [reply site/ja/tutorials/customization/eager/custom_training_walkthrough.ipynb]: Thanks.  Deleted.

**Comment by AseiSugiyama** [comment]
LGTM!

**Comment by masa-ita** [comment]
Oops. I forgot the disclaimer.  Thanks @lamberta .
Thanks all reviews.

**Pull request merged**